ceno_zkvm/src/instructions/riscv/mul.rs

```diff
@@ -308,16 +350,32 @@ impl<E: ExtensionField, I: RIVInstruction> Instruction<E> for MulhInstructionBas
                 rd_signed.assign_instance(instance, lk_multiplicity, &rd_val)?;
 
                 let prod = (rs1_s as i64) * (rs2_s as i64);
-                ((prod as u64) % (1u64 << BIT_WIDTH)) as u32
+                // only take the low part of the product
+                prod as u32
```

Comment:
Btw, this is just `rs1.wrapping_mul(rs2);`.  No need to futz around with the signs.

Reply:
addressed in [11df823](https://github.com/scroll-tech/ceno/pull/660/commits/11df82393415723d06aab78410cf4d8665f76254)